packages/blade/src/components/Input/PaymentInput/_decisions/decisions.md

```diff
@@ -0,0 +1,221 @@
+# PaymentInput Decisions <!-- omit in toc -->
+
+PaymentInput extends existing TextInput and PasswordInput components with formatting capabilities for payment-related data including card numbers, expiry dates, and CVV codes.
+
+<img src="./payment-input-anatomy.png" width="380" alt="PaymentInput component anatomy" />
+
+- [Design](#design)
+- [API](#api)
+  - [TextInput with Formatting](#textinput-with-formatting)
+  - [CVV Input (PasswordInput)](#cvv-input-passwordinput)
+- [Usage Examples](#usage-examples)
+- [Decisions](#decisions)
+- [Open Questions](#open-questions)
+- [References](#references)
+
+## Design
+
+[Figma Link](https://www.figma.com/design/jubmQL9Z8V7881ayUD95ps/Blade-DSL?node-id=10953-185841&p=f&m=dev) to all variants of the PaymentInput component
+
+## API
+
+### TextInput with Formatting
+
+We extend existing TextInput with formatting capabilities using a generic approach:
+
+```ts
+type TextInputProps = BaseInputProps & {
+  /**
+   * Format pattern where # represents digits and other characters are delimiters
+   * When provided, input will be automatically formatted
+   * @example "#### #### #### ####" for card numbers
+   * @example "##/##" for expiry dates
+   * @example "(###) ###-####" for phone numbers
+   */
+  format?: string;
+
+  /**
+   * Trailing React component (e.g., Image, Icon)
+   * When provided, component will be shown at the end of the input
+   */
+  trailing?: React.ComponentType<any>;
+
+  /**
+   * Enhanced onChange provides both formatted and raw values
+   */
+  onChange?: ({
+    value: string; // Formatted value (e.g., "1234 5678 9012 3456")
+    rawValue: string; // Raw digits only (e.g., "1234567890123456")
+    name?: string;
+  }) => void;
+};
+```
+
+### CVV Input (PasswordInput)
+
+For CVV, we keep it simple with just character limits:
+
+```ts
+type CVVInputProps = BaseInputProps & {
```

Comment:
Let's call this PasswordInputProps, this gives an impression that we will be creating a new subcomponent.